oxidation/libparsec/crates/platform_local_db/src/executor_test.rs

```diff
@@ -0,0 +1,237 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use crate::DatabaseError;
+use diesel::{connection::SimpleConnection, Connection, SqliteConnection};
+use rstest::rstest;
+use std::sync::{Arc, Mutex};
+
+use super::SqliteExecutor;
+
+#[rstest]
+#[test_log::test(tokio::test)]
+async fn dropped_caller() {
+    let connection = SqliteConnection::establish(":memory:").unwrap();
+    let executor = SqliteExecutor::start(connection, Ok);
+
+    let events: Arc<Mutex<Vec<u32>>> = Arc::new(Mutex::new(vec![]));
+
+    // We ask for a job...
+    let job_handle = executor
+        .submit({
+            let events = events.clone();
+            move |_conn| {
+                events.lock().unwrap().push(1);
+                1
+            }
+        })
+        .await
+        .unwrap();
+
+    // ...then we forget about the result
+    drop(job_handle);
+
+    // We can then ask for another job without trouble
+    let result = executor
+        .exec({
+            let events = events.clone();
+            move |_conn| {
+                events.lock().unwrap().push(2);
+                2
+            }
+        })
+        .await
+        .unwrap();
+    assert_eq!(result, 2);
+
+    // Also test vacuum
+    let job_handle = executor.submit_full_vacuum().await;
+    // ...then forget about the result !
+    drop(job_handle);
+
+    // Again, the executor should still be working
+    executor.full_vacuum().await.unwrap();
+    let result = executor
+        .exec({
+            let events = events.clone();
+            move |_conn| {
+                events.lock().unwrap().push(3);
+                3
+            }
+        })
+        .await
+        .unwrap();
+    assert_eq!(result, 3);
+
+    // Final check: make sure all jobs got executed in order, including the one that
+    // got it handle dropped
+    assert_eq!(events.lock().unwrap().as_ref(), [1, 2, 3]);
+}
+
+#[rstest]
+#[test_log::test(tokio::test)]
+async fn panic_with_multiple_jobs() {
+    let connection = SqliteConnection::establish(":memory:").unwrap();
+    let executor = Arc::new(SqliteExecutor::start(connection, Ok));
+
+    // First job is here to make the background executor on hold and make sure
+    // the panic job won't be executed before the ok jobs is also scheduled
+    let barrier_job = executor
+        .submit(move |_conn| {
+            // Here we should be waiting until panic & ok jobs are submitted, however
+            // there is no easy way to know when this time has come: the job queue being
+            // unbounded, submit will block until the job has started execution.
+            // So instead we play the sleep card :(
+            std::thread::sleep(std::time::Duration::from_millis(10));
+        })
+        .await
+        .unwrap();
+
+    let mut join_set = libparsec_platform_async::JoinSet::new();
```

Comment:
```suggestion
    let mut join_set = libparsec_platform_async::JoinSet::default();
```

Reply:
`::default()` doesn't work if the type of the future has a return value which doesn't itself implement default

Reply:
![image](https://user-images.githubusercontent.com/3187637/226571889-9e22123a-cc64-40ba-80f4-b8d9574d954c.png)